### ISSUE TITLE
[Jefferson Peak] Bluetooth LE discovery broken on latest SFI

### ISSUE THREAD

**Issue by Sean-StarLabs** [StarLabsLtd/firmware#180]
SFIs newer than 20420220 can't discover LE devices.

Can fix with the below:

```
wget https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-20240220.tar.gz
sudo tar -xzvf linux-firmware-20240220.tar.gz --strip-components=2 --directory /usr/lib/firmware/intel linux-firmware-20240220/intel/ibt-0040-2120.sfi linux-firmware-20240220/intel/ibt-0040-2120.ddc
```

**Comment by EverlastingOS** [comment]
Have tried, same issue, can connect to my airpods but not the surface mouse.

**Comment by sodiboo** [comment]
I'm also experiencing this same issue. On NixOS, with this patch (https://github.com/sodiboo/nix-config/commit/c5b735eec853fac5c84ea34bf255f6b740584993) applied followed by a reboot, the issue persists. Still cannot connect my mouse.

**Comment by sodiboo** [comment]
I also noticed in #175, one stated change is

> Improved Bluetooth compatibility

So i updated my BIOS to the latest version following those instructions. It went well, but didn't solve this issue. Still cannot connect my mouse.

**Comment by rogercreagh** [comment]
Same here but Bluetooth 3 mouse (actually mice) works ok after the BIOS update.

**Comment by edpenz** [comment]
Fixed my Logitech G604 mouse too.

I also needed to do a `sudo update-initramfs -r` and reboot for it to take effect on Ubuntu 24.04.

**Comment by sodiboo** [comment]
@EverlastingOS What distro do you use? It's interesting that the fix works for several others, but not for me or for you. NixOS *is* weird and the installation instructions don't work outright, so in that case my failure might be because i'm doing it wrong or something. But it doesn't work for you. Are you also using NixOS or a similar non-standard distro that might cause these issues, or were you able to just straight up run the suggested commands, but it still didn't work?

**Comment by EverlastingOS** [comment]
> @EverlastingOS What distro do you use? It's interesting that the fix works for several others, but not for me or for you. NixOS _is_ weird and the installation instructions don't work outright, so in that case my failure might be because i'm doing it wrong or something. But it doesn't work for you. Are you also using NixOS or a similar non-standard distro that might cause these issues, or were you able to just straight up run the suggested commands, but it still didn't work?

Hey, i have tried with Fedora 40 and Elementary OS.

**Comment by rbm78bln** [comment]
> Have tried, same issue, can connect to my airpods but not the surface mouse.

Can confirm. Discovering the surface mouse still doesn't work with neither firmware nor BIOS update. (Distro is Arch)

Nevertheless connecting to an already paired BLE device does work indeed: I had the bd address of a CSR Bluetooth device (RasPi3b+) [temporarily changed](https://github.com/bluez/bluez/blob/master/tools/bdaddr.rst) to the address of my StarLite, paired my surface mouse with it, and transferred the relevant files and keys from /var/lib/bluetooth from the "donor device" back to my StarLite. Now as a result both know each other's keys and addresses and I can use my mouse nicely. This also confirms that it's basically only a discovery issue.

Maybe this will also do as a temporary workaround for you guys (@EverlastingOS, @sodiboo, etc.)

**Comment by sodiboo** [comment]
oh wow i did not even think of using a separate device to pair them! haha that sounds like a great hacky workaround, thanks @rbm78bln

**Comment by EverlastingOS** [comment]
> > Have tried, same issue, can connect to my airpods but not the surface mouse.
> 
> Can confirm. Discovering the surface mouse still doesn't work with neither firmware nor BIOS update. (Distro is Arch)
> 
> Nevertheless connecting to an already paired BLE device does work indeed: I had the bd address of a CSR Bluetooth device (RasPi3b+) [temporarily changed](https://github.com/bluez/bluez/blob/master/tools/bdaddr.rst) to the address of my StarLite, paired my surface mouse with it, and transferred the relevant files and keys from /var/lib/bluetooth from the "donor device" back to my StarLite. Now as a result both know each other's keys and addresses and I can use my mouse nicely. This also confirms that it's basically "only" a discovery issue.
> 
> Maybe this will also do as a temporary workaround for you guys (@EverlastingOS, @sodiboo, etc.)

Did copy the files for my mouse, put it onto the starlite device, so can see that the mouse it there in the list but cannont connect, did try connect with CLI, "got Failed to pair: org.bluez.Error.Failed"

**Comment by rbm78bln** [comment]
> got Failed to pair: org.bluez.Error.Failed

Are you trying to pair the mouse with the starlite again? After transferring the pairing keys to the tablet, I just used "connect xx:xx:xx...." to connect to the mouse from the tablet on the bluetoothctl-command line. While doing the nasty stuff, additionally I made sure that all involved devices had bluetooth turned off completely and also rebooted all hosts afterwards.

**Comment by EverlastingOS** [comment]
> > got Failed to pair: org.bluez.Error.Failed
> 
> Are you trying to pair the mouse with the starlite again? After transferring the pairing keys to the tablet, I just used "connect xx:xx:xx...." to connect to the mouse from the tablet on the bluetoothctl-command line. While doing the nasty stuff, additionally I made sure that all involved devices had bluetooth turned off completely and also rebooted all hosts afterwards.

Do get "Failed to connect: org.bluez.Error.Failed le-connection-abort-by-local". Do you mod something in the files from your raspi device? I just copied the two files "attributes" and "Info", did you change anything in them as key or so?

**Comment by rbm78bln** [comment]
> Do you mod something in the files from your raspi device?

Nope, I didn't change the files, but intermittently I had changed the hardware MAC address of the raspi to match one from my starlite. Then, after copying the files and rebooting the Starlite, I didn't try pairing again, but just used the mouse.

**Comment by EverlastingOS** [comment]
Still having this issue even with  24.0710 release

**Comment by Sean-StarLabs** [comment]
Nothing in 24/07/10 relating to Bluetooth - should get some stuff back from Intel tomorrow to hopefully fix

**Comment by EverlastingOS** [comment]
oh okey, great to know :) Lets see if they can give us a fix :)

**Comment by Sean-StarLabs** [comment]
Got a fix. Complexity of implementing said fix into coreboot is probably 312 out of 10, so might take a few days :/

**Comment by rogercreagh** [comment]
Bluetooth 5.1 keyboard (see discussion https://github.com/StarLabsLtd/firmware/discussions/204) is detected and connecting fine with firmware 24.07. Bluetooth 5.0 mouse still not detected although 3.0 is (as previously)

**Comment by MRDGH2821** [comment]
I updated my firmware to 24.07 - no mouse detected
I did the steps as mentioned - no mouse detected
`sudo update-initramfs -u` - no mouse detected

After each step I rebooted & tried connecting to mouse, same result - no mouse detected.

I'm using [Logitech Vertical Ergonomic Mouse](https://www.logitech.com/en-us/products/mice/lift-vertical-ergonomic-mouse.910-006466.html) and the bluetooth mode is not detected by my tablet

**Comment by Sean-StarLabs** [comment]
Fixed with 00bd1d2ebfecc59f98d72892159bdee07ae12d36

**Comment by jothgard** [comment]
> Fixed with [00bd1d2](https://github.com/StarLabsLtd/firmware/commit/00bd1d2ebfecc59f98d72892159bdee07ae12d36)

Seems still not find my Microsoft mice on Ubuntu 24.04. Did I miss something?

**Comment by Sean-StarLabs** [comment]
> Did I miss something?

How did you install it?

**Comment by stronnag** [comment]
```
$ cat /sys/class/dmi/id/bios_version
24.08
```
Still fails to discover **any** BLE devices.

**Comment by Sean-StarLabs** [comment]
> Still fails to discover any BLE devices.

How did you install it?

**Comment by stronnag** [comment]
Everything else looks great (Battery getting above 97% at last, no more annoying keyboard backlight flashes). Just no BLE discovery for 2 x serial devices and 1 Omcon BP monitor.

Arch Linux with LQX 10.9 kernel.

**Comment by jothgard** [comment]
> `sudo fwupdmgr --force install /tmp/coreboot-I5.cab`

Trying to install, do only get :
No supported devices found

@Sean-StarLabs How can I install oreboot-I5.cab manually? Do see that the EC version is still on 24.07 and not updated to 24.08 as the Bios version.

**Comment by Sean-StarLabs** [comment]
You can't manually update an EC, it'll update automatically when you connect a charger in G3

**Comment by jothgard** [comment]
> You can't manually update an EC, it'll update automatically when you connect a charger in G3

Do you med i need to have the charger connected to the starlite 5 while typing: flashrom -p internal -w 24.08.rom  -i bios --ifd -n -N

**Comment by Sean-StarLabs** [comment]
No - it's automatic, i.e. you don't do anything

Just turn if off, wait for 12 seconds so it's in G3, then connect a charger.

**Comment by jothgard** [comment]
> Just turn if off, wait for 12 seconds so it's in G3, then connect a charger.

Oh okey, yeah, did just do this and sees that im in 24.08 on EC as well now, but cannot connect to the surface mouse.

**Comment by adamtclark** [comment]
Same issue for me - my bluetooth keyboard (Logitech Keys-to-Go 2) doesn't show up in the manager as visible on my Starlite Mk V (tried both via GUI and bluetoothctl, running Ubuntu 24.04.1 LTS, firmware version 24.08 installed via LVFS). The keyboard connects no problem on every other device I own (other Ubuntu computers, Android phone), so seems to be an issue with the Mk V. Any suggestions on how to proceed?

**Comment by adamtclark** [comment]
And just for completeness - I can confirm that I am able to connect to the keyboard on my Mk V without problems through a bluetooth dongle (TP-link UB500), after running:
sudo modprobe btusb
sudo systemctl restart bluetooth.service

and unplugging/replugging the dongle, as described here: https://askubuntu.com/questions/1423752/ubuntu-20-04-unable-to-use-a-bluetooth-dongle-tp-link-ub500

**Comment by srund** [comment]
Same issue; BLE devices are not discovered on my Starlite. Annoying since I work with BLE devices daily. The possible fix ([here](https://github.com/StarLabsLtd/firmware/issues/180#issue-2383781373)) didn't resolve the issue.

Running Ubuntu 24.04  as installed by Starlabs, updated to kernel 6.8.0-45-generic and using BIOS 24.08/24.8 as updated with `fwupd` .

System logs look reasonable to me (except the lack of BLE events). Even when Bluez log on debug level. Sniffed usb traffic to the wireless chip look somewhat fishy but I need compare with a working device before saying too much.

**Comment by Sean-StarLabs** [comment]
Fixed with d6de2690dd547a014309086a9cc435e2b58f091e

**Comment by jothgard** [comment]
> Fixed with [d6de269](https://github.com/StarLabsLtd/firmware/commit/d6de2690dd547a014309086a9cc435e2b58f091e)

Thanks! How can i download the latest? did try with wget for 24.09, seems not to download the rom.

**Comment by Sean-StarLabs** [comment]
LVFS - or download the rom from the repo:
https://github.com/StarLabsLtd/firmware/tree/main/StarLite/MkV/coreboot/24.09

**Comment by jothgard** [comment]
> LVFS - or download the rom from the repo: https://github.com/StarLabsLtd/firmware/tree/main/StarLite/MkV/coreboot/24.09

trying to download the link above with wget, but the rom file is only some kb, should be 20 mb+?

**Comment by MRDGH2821** [comment]
Flashed 24.09, but bluetooth mouse & my android phone still not detected.
I also see coreboot showing 0 MB RAM.

**Comment by jothgard** [comment]
> Flashed 24.09, but bluetooth mouse & my android phone still not detected. I also see coreboot showing 0 MB RAM.
> 
> Update: I downgraded to 24.08. On 24.09, CPU usage hit 100% almost everytime I opened Firefox or VS Code. And the system becomes almost unresponsive for few seconds. In 24.08, CPU usage rarely ever goes 100% & the system is responsive even at 90%+ CPU usage.

yes, same here, thats why my starlite is crazy slow on 24.09 even om boot up..

**Comment by Sean-StarLabs** [comment]
What kernels and SFI's please?

24.09 on max benchmarks 20% higher than 24.08, so is this just a "first notice and bail out"?

**Comment by MRDGH2821** [comment]
> What kernels and SFI's please?

No idea about SFI. I have reinstalled various OSes in my tabnlet, so I'm assuming latest SFI provided by OS (currently on [Endeavour OS](https://endeavouros.com/))
[Liquorix Kernel](https://liquorix.net/) v6.11.2 & v6.11.3

> 24.09 on max benchmarks 20% higher than 24.08, so is this just a "first notice and bail out"?
I stayed on 24.09 for 2-3 days before downgrading.

**Comment by Sean-StarLabs** [comment]
Okay, can you chuck the latest ones on:

```
wget https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-20241017.tar.gz
sudo tar -xzvf linux-firmware-20241017.tar.gz --strip-components=2 --directory /usr/lib/firmware/intel linux-firmware-20240220/intel/ibt-0040-2120.sfi linux-firmware-20240220/intel/ibt-0040-2120.ddc
```

> I stayed on 24.09 for 2-3 days before downgrading.

Default firmware settings and P-State (assuming Endeavour sets the P-State)?

**Comment by MRDGH2821** [comment]
I fixed the date than ran the commands.
```
wget https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git/snapshot/linux-firmware-20241017.tar.gz
sudo tar -xzvf linux-firmware-20241017.tar.gz --strip-components=2 --directory /usr/lib/firmware/intel linux-firmware-20241017/intel/ibt-0040-2120.sfi linux-firmware-20241017/intel/ibt-0040-2120.ddc
```

After rebooting my [tablet](https://starlabs.systems/pages/starlite), [phone](https://in.nothing.tech/pages/phone-1), & [computer](https://www.gigabyte.com/us/Motherboard/X670-AORUS-ELITE-AX-rev-10-12) and switching off-on my [bluetooth keyboard](https://amzn.in/d/bV09yXF) & [wireless mouse](https://www.logitech.com/en-us/products/mice/lift-vertical-ergonomic-mouse.html), mobile-tablet-pc were interconnected with each other correctly without any problems. 
Keyboard was now discoverable by tablet & successfully connected too. Only the mouse was not able to connect.

> Default firmware settings and P-State (assuming Endeavour sets the P-State)?

I changed charging rate to `1.0 C` in BIOS settings. Apart from that I did not change anything else. 
Don't know about the P-State. 
Endeavour OS is arch based, so I'm assuming whatever default Arch is setting, E-OS is also using the same.

**Comment by Sean-StarLabs** [comment]
> Only the mouse was not able to connect.

How about with 24.09? If not, what's the mouse?

> I'm assuming whatever default Arch is setting, E-OS is also using the same.

Probably - that's just balanced which means the firmware settings have much more impact. If you want more performance, just a case of turning them up.

**Comment by MRDGH2821** [comment]
> How about with 24.09? If not, what's the mouse?

Flashed 24.09, mouse is not detected.
It is Logitech Lift Vertical Ergonomic Mouse - https://www.logitech.com/en-us/products/mice/lift-vertical-ergonomic-mouse.html 

I'll observe my tablet for few days in 24.09 firmware in regards to performance.

**Comment by bergmannf** [comment]
I finished the 24.09 update as well, but it's still not detecting a Razer Basilik X bluetooth mouse.
On top of that I can confirm the performance issues already mentioned earlier that make the tablet feel very sluggish (same issue with it saturating CPUs - so I also opted to downgrade, as it feels really slow) - I've opened a new issue #221  for what feel like performance regression, to maybe not sidetrack this one too much.

**Comment by Sean-StarLabs** [comment]
Ha - that's one of the ones I tested with. Bring the latest SFI's in as above.

**Comment by bergmannf** [comment]
Hm I tried that on Fedora, but it didn't work - I am not sure I did it entirely correct though, as the firmware files on that distribution seem to be `xz` compressed - so I tried it without compressed files *and* compressed files (and rebuild the initram with `dracut -f` but neither worked).

I am also not sure if I should even need to install any upstream files, given that the package shipped is already the firmware files from 17.10:

```
Name         : linux-firmware
Version      : 20241017
```

And looking at the installed files from that package shows that the ones extracted from the tar files are symlinks:

```
lrwxrwxrwx. 1 root root 20 Oct 19 02:00 /usr/lib/firmware/intel/ibt-0040-2120.ddc.xz -> ibt-0040-0041.ddc.xz
lrwxrwxrwx. 1 root root 20 Oct 19 02:00 /usr/lib/firmware/intel/ibt-0040-2120.sfi.xz -> ibt-0040-1020.sfi.xz
```

**Comment by jothgard** [comment]
running 24.10, still cannot find my Microsoft mouse. Using Microsoft Modern Mobile Mouse.

**Comment by jothgard** [comment]
> 24.10 hasn't even been tested yet, so it'll just complicate things. Please post the SFI versions and kernel

Running Fedora silverblue on kernel 6.11.3-200.fc40.x86_64, how can i see what SFI version im on?

**Comment by Sean-StarLabs** [comment]
`sudo dmesg  | grep hci0`

**Comment by jothgard** [comment]
> sudo dmesg | grep hci0

got output:

[   27.803755] Bluetooth: hci0: Device revision is 2
[   27.803760] Bluetooth: hci0: Secure boot is enabled
[   27.803761] Bluetooth: hci0: OTP lock is enabled
[   27.803762] Bluetooth: hci0: API lock is enabled
[   27.803763] Bluetooth: hci0: Debug lock is disabled
[   27.803764] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[   27.803765] Bluetooth: hci0: Bootloader timestamp 2019.40 buildtype 1 build 38
[   27.803829] Bluetooth: hci0: DSM reset method type: 0x00
[   27.845244] Bluetooth: hci0: Found device firmware: intel/ibt-0040-2120.sfi
[   27.845261] Bluetooth: hci0: Boot Address: 0x100800
[   27.845263] Bluetooth: hci0: Firmware Version: 151-5.24
[   29.378012] Bluetooth: hci0: Waiting for firmware download to complete
[   29.378716] Bluetooth: hci0: Firmware loaded in 1497525 usecs
[   29.378755] Bluetooth: hci0: Waiting for device to boot
[   29.395737] Bluetooth: hci0: Malformed MSFT vendor event: 0x02
[   29.395742] Bluetooth: hci0: Device booted in 16606 usecs
[   29.400525] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0040-2120.ddc
[   29.402750] Bluetooth: hci0: Applying Intel DDC parameters completed
[   29.406429] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151
[   29.406437] Bluetooth: hci0: Firmware SHA1: 0x00000000
[   29.409967] Bluetooth: hci0: Fseq status: Success (0x00)
[   29.409973] Bluetooth: hci0: Fseq executed: 00.00.02.36
[   29.409975] Bluetooth: hci0: Fseq BT Top: 00.00.02.36

**Comment by adamtclark** [comment]
Still same problem for me - BLE keyboard and mouse work fine if I connect them over a USB bluetooth dongle, but don't work at all using the internal Bluetooth of my MkV. Output from dmsg is pasted below - I'm still running Ubuntu 24.04.01.

[    3.848780] Bluetooth: hci0: Device revision is 2
[    3.848786] Bluetooth: hci0: Secure boot is enabled
[    3.848787] Bluetooth: hci0: OTP lock is enabled
[    3.848788] Bluetooth: hci0: API lock is enabled
[    3.848789] Bluetooth: hci0: Debug lock is disabled
[    3.848790] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[    3.848792] Bluetooth: hci0: Bootloader timestamp 2019.40 buildtype 1 build 38
[    3.848886] Bluetooth: hci0: DSM reset method type: 0x00
[    3.851894] Bluetooth: hci0: Found device firmware: intel/ibt-0040-2120.sfi
[    3.851913] Bluetooth: hci0: Boot Address: 0x100800
[    3.851914] Bluetooth: hci0: Firmware Version: 151-5.24
[    5.387316] Bluetooth: hci0: Waiting for firmware download to complete
[    5.387549] Bluetooth: hci0: Firmware loaded in 1499656 usecs
[    5.387613] Bluetooth: hci0: Waiting for device to boot
[    5.404547] Bluetooth: hci0: Device booted in 16564 usecs
[    5.404561] Bluetooth: hci0: Malformed MSFT vendor event: 0x02
[    5.405741] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0040-2120.ddc
[    5.407565] Bluetooth: hci0: Applying Intel DDC parameters completed
[    5.410578] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151
[    5.410584] Bluetooth: hci0: Firmware SHA1: 0x00000000

**Comment by rbm78bln** [comment]
Yes, same for me. Also booting takes considerably longer until the device is finally up with 24.09.

Before I had a workaround running to get me over the ble pairing hurdle, now even already the paired mouse doesn't connect properly anymore. Actually the mouse connects for a short period time, service resolution seems to fail and then it is disconnected again.

I have created a few logs including versions and trances and everything. Please find them attached.

BTW: Wouldn't it make sense to reopen this ticket again? Clearly this doesn't seem to be solved.


[blebug_bluetoothctl.txt](https://github.com/user-attachments/files/17611054/blebug_bluetoothctl.txt)
[blebug_dmesg.txt](https://github.com/user-attachments/files/17611055/blebug_dmesg.txt)
[blebug_hcitrace.txt](https://github.com/user-attachments/files/17611056/blebug_hcitrace.txt)
[blebug_syslog.txt](https://github.com/user-attachments/files/17611057/blebug_syslog.txt)
[blebug_hcitrace_pcapng_base64.txt](https://github.com/user-attachments/files/17611062/blebug_hcitrace_pcapng_base64.txt)

**Comment by stronnag** [comment]
Apropos boot time, you are probably waiting for the non-appearing tpm device. May be resolved by:

```
sudo systemctl mask dev-tpmrm0.device
```

**Comment by rbm78bln** [comment]
> you are probably waiting for the non-appearing tpm device

Naah, was fixing this mid-year already... 😎

**Comment by bergmannf** [comment]
Just as an update - it seems the new Bios 24.10 did not help with this issue either:
```
sudo dmidecode -s bios-version
24.10
```

And the HCI info:

```
[   49.073343] Bluetooth: hci0: Device revision is 2
[   49.073352] Bluetooth: hci0: Secure boot is enabled
[   49.073353] Bluetooth: hci0: OTP lock is enabled
[   49.073354] Bluetooth: hci0: API lock is enabled
[   49.073354] Bluetooth: hci0: Debug lock is disabled
[   49.073355] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[   49.073357] Bluetooth: hci0: Bootloader timestamp 2019.40 buildtype 1 build 38
[   49.073419] Bluetooth: hci0: DSM reset method type: 0x00
[   49.116960] Bluetooth: hci0: Found device firmware: intel/ibt-0040-2120.sfi
[   49.116982] Bluetooth: hci0: Boot Address: 0x100800
[   49.116984] Bluetooth: hci0: Firmware Version: 151-5.24
[   50.643101] Bluetooth: hci0: Waiting for firmware download to complete
[   50.643424] Bluetooth: hci0: Firmware loaded in 1490686 usecs
[   50.643457] Bluetooth: hci0: Waiting for device to boot
[   50.660353] Bluetooth: hci0: Malformed MSFT vendor event: 0x02
[   50.660356] Bluetooth: hci0: Device booted in 16519 usecs
[   50.663812] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0040-2120.ddc
[   50.665385] Bluetooth: hci0: Applying Intel DDC parameters completed
[   50.668370] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151
[   50.668378] Bluetooth: hci0: Firmware SHA1: 0x00000000
[   50.672355] Bluetooth: hci0: Fseq status: Success (0x00)
[   50.672363] Bluetooth: hci0: Fseq executed: 00.00.02.36
[   50.672365] Bluetooth: hci0: Fseq BT Top: 00.00.02.36
```

Still won't detect LE devices.

**Comment by Sean-StarLabs** [comment]
>  Just as an update - it seems the new Bios 24.10 did not help with this issue either:

Nothing in there that would

> [   50.668370] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151

The fix is in the week 33 builds,  so you'll need to manually update those or wait for you distro to get the newer firmware.

**Comment by altatechor** [comment]
> The fix is in the week 33 builds

Where can I find those, and how would I update manually? With `flashrom`?

**Comment by thebatdan** [comment]
I'm having no luck connecting bluetooth mice. I've tried two different models, Dell MS5120W and Logitech G604, neither will connect. Updating the SFI makes no difference.

Running Fedora 41 with firmware version 24.10

**Comment by maltklaus** [comment]
Same issue here, fully patched Manjaro (testing branch) and can't get it to connect. I'm on firmware 24.11 which is the version the device  (Starlite V) shipped with (just got it a week ago...).

**Comment by rbm78bln** [comment]
Yes, still no luck...

> I'm on firmware 24.11 which is the version the device (Starlite V) shipped with

Nice, @maltklaus! I'm wondering when it's gonna be [released](https://github.com/StarLabsLtd/firmware/tree/main/StarLite/MkV/coreboot).

**Comment by Sean-StarLabs** [comment]
Fixed with 457d8dd513c9ae57f56bb0e1fec96c91f91c598b

**Comment by stronnag** [comment]
Sorry to report still not fixed here with 25.01 firmware (Arch Linux testing, latest linux-firmware).
However, Legacy BT discovery does seem more reliable.
```
[    6.507458] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151

$ sudo dmidecode -s bios-version
25.01
$ uname -a
Linux piglet 6.13.0-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Thu, 23 Jan 2025 01:21:06 +0000 x86_64 GNU/Linux
```

**Comment by stronnag** [comment]
> Did you corrupt your checksums, and reboot after exiting G3 with a charger?

Not that I'm aware of. Followed your prior instructions to flash, including the shutdown / remove power / wait for leds / add power again dance. 

And (briefly) expected that this time, it really would discover BLE devices. But it doesn't.

**Comment by maltklaus** [comment]
well, sh....
didn't find the udate for the official procedure. Spent quite some time trying to find info on how to update manually. Ended up using flashrom as posted earlier with the whole G3 wait thingy. Confirmed Coreboot and EFI on 25.01
Things are now worse than before. No BLE for my Logitech M240 AND my battery reports a constant level of 0% (it was at 96% when I did the update). Guess I now have a paperweight instead of the new "just working" working machine I hoped to get when I ordered it in November...

**Comment by maltklaus** [comment]
downgraded to 24.10 using flashrom, confirmed I am indeed on that version. No change to the battery showing 0%. :(

**Comment by Sean-StarLabs** [comment]
> And (briefly) expected that this time, it really would discover BLE devices. But it doesn't.

Cool - not sure even if you did corrupt checksums, what the exact steps would be to get it in the right state - by the time I do, the LVFS will have caught up so just do as reinstall

> didn't find the udate for the official procedure

This is what's meant by "peril" - you've flashed the wrong binary! Ping support with your order number, and link to this issue, should be able to sort it out

**Comment by maltklaus** [comment]
> > didn't find the udate for the official procedure
> 
> This is what's meant by "peril" - you've flashed the wrong binary! Ping support with your order number, and link to this issue, should be able to sort it out

I could only find one binary that was linked here - I did contact support as instructed, thanks!

**Comment by thebatdan** [comment]
I've updated to firmware 25.01 from LVFS. Still no luck connecting either of my two bluetooth mice to the starlite. Both connect instantly to other devices.

The 25.01 update was applied using fwups on Fedora 41. Any special instructions that need to be followed to get the Bluetooth working properly?

**Comment by Sean-StarLabs** [comment]
Seems inconstant - will add some bits to the next release to force it into the correct state

**Comment by stronnag** [comment]
> Seems inconstant - will add some bits to the next release to force it into the correct state

Thank you Sean. I still have the same issue with the latest (25.01) via fwupd.

**Comment by adamtclark** [comment]
Likewise - I updated to 25.01 using the Ubuntu "Firmware Update" tool, and now my battery is showing 0% after shutdown / remove power / wait for leds / add power again. And BLE is still not working.

**Comment by tbocek** [comment]
I also updated Star Lite Mk V on Arch to 25.01(https://github.com/fwupd/fwupd/issues/8251#issuecomment-2633519118), and I cannot connect my Bluetooth devices either. Bluetooth was running with 24.10 and kernel 6.6, but now Arch changed lts to 6.12, where I had the screen flickering issue, so I upgraded to 25.01. With 25.01, I have no luck connection my BT keyboard and mouse.

**Comment by jothgard** [comment]
How can i update to 25.01 on fedora silverblue?

**Comment by jothgard** [comment]
> I also updated Star Lite Mk V on Arch to 25.01([fwupd/fwupd#8251 (comment)](https://github.com/fwupd/fwupd/issues/8251#issuecomment-2631704809)), and I cannot connect my Bluetooth devices either. Bluetooth was running with 24.10 and kernel 6.6, but now Arch changed lts to 6.12, where I had the screen flickering issue, so I upgraded to 25.01. With 25.01, I have no luck connection my BT keyboard and mouse.

Did also update to 25.01, same issue, cant find my bluetooth mice.

**Comment by ssievert42** [comment]
I've been trying to connect to Espruino devices (Microcontrollers that talk over BLE and can run JS (which makes rapid iteration super easy)) with no luck.

Also on the latest 25.01 via fwupd, Ubuntu 24.04, Linux 6.8.0 and the linux-firmware that Ubuntu ships with (currently 2024-03-18).
To update, I've used the gnome-firmware GUI, plugged in a charger, selected coreboot 25.01, waited for it to install, powered down the StarLite, unplugged the charger, waited for the power LED to flicker rapidly after 12 seconds, reconnected the charger and started the StarLite again.
The coreboot GUI shows 25.01 as both the coreboot and EC version and `/sys/class/dmi/id/bios_version` and `/sys/class/dmi/id/ec_firmware_release` show the latest version as well.
So I'm definitely on 25.01 :)

Is that manually fetching newer firmware from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git still gonna be needed once a fix lands?

To check for BLE devices that my StarLite Mk V can see, I've been doing:
```bash
bluetoothctl
scan le
```
and then periodically typing `devices`.
On my laptop, this eventually prints something along the lines of:
```
[bluetooth]# [NEW] Device CA:FE:DE:AD:BE:EF DeviceNameHere
```
and lists that device when typing `devices`, but on the StarLite Mk V nothing happens.

My "regular bluetooth"-headphones still work fine though.

Thanks for taking another look into this!

**Comment by maltklaus** [comment]
> Is that manually fetching newer firmware from https://git.kernel.org/pub/scm/linux/kernel/git/firmware/linux-firmware.git still gonna be needed once a fix lands?
> 

I'm on Arch with the very latest Linux firmware and it's the same here so I wouldn't bother with it as an attempted fix unless instructed specifically by Starlabs.

**Comment by sosukeinu** [comment]
I've resisted posting here, as mine would just be another voice saying the same thing. I'm on Ubuntu Noble. I was using fw 24.04. Under 24.04 I was able to connect 2 separate Bluetooth keyboards without issue, but was never able to get a mouse working. Upgraded to fw 25.01 today, and now the keyboards refuse to work. I'm still getting connection with the keyboard, but pressing keys does nothing. It was working minutes before the upgrade.

**Comment by thebatdan** [comment]
@Sean-StarLabs any updates on this issue? When can we expect to be able to use Bluetooth devices with the Starlight?

**Comment by rogercreagh** [comment]
seems to me there are too many variables in this problem - not only the firmware loaded and OS type, version and config, but also the devices and combination of devices in use.

I have Ubuntu 24.04.1 kept up to date with updates, but am sticking with Firmware 24.10 because it works with my particular bluetooth mouse, keyboard and other devices. 

I've been ignoring notices about firmware updates for fear of breaking a working system - I suspect for every particular device that starts working with newer firmware another existing working device will fail.

This is like trying to nail down mercury. The only slight problem I have is that with my particular keyboard (Arteck HB206-2) the caps lock sometimes gets inverted when the system reawakens after a sleep - but I can live with that.

Perhaps a list of working combinations would be helpful?

My starter for 10 - Firmware 24.10 on Ununtu 24.04.1 works with Arteck HB206-2 keyboard and Kensington Pro Fit Bluetooth Mouse (model number illegible but possibly 902-0102-02)

**Comment by jothgard** [comment]
Did just buy a new BT logitech mouse, does not work either, so really hoping for at fix soon..

**Comment by jpritikin** [comment]
Firmware 25.02 was just released. There is mention of bluetooth in the release notes. Has anybody tested whether LE is fixed?

**Comment by tbocek** [comment]
Tested my BT keyboard, but no luck with 25.02. Both, coreboot and EC are on 25.02, still cant find my keyboard

**Comment by Sean-StarLabs** [comment]
> Firmware 25.02 was just released.

No, it wasn't - https://fwupd.org/lvfs/devices/com.starlabs.I5.coreboot

I'll post when is it, along with some of the versions on the linux side that has bugs

**Comment by tbocek** [comment]
What, I just downloaded https://github.com/StarLabsLtd/firmware/tree/main/StarLite/MkV/coreboot/25.02, cabextracted it, then ```sudo flashrom -p internal -w 25.02.rom -i bios --ifd -n -N```, and my bios says 25.02. 

How is it not released?

**Comment by jothgard** [comment]
> > Firmware 25.02 was just released.
> 
> No, it wasn't - https://fwupd.org/lvfs/devices/com.starlabs.I5.coreboot
> 
> I'll post when is it, along with some of the versions on the linux side that has bugs

should or can we manually install .cab file?

**Comment by Sean-StarLabs** [comment]
> How is it not released?

It's  "released" when it gets marked as stable in the LVFS. It needs final checks, and to be installed with fwupd, because not everyone has programmers for when things go wrong with flashrom

**Comment by jothgard** [comment]
> What, I just downloaded https://github.com/StarLabsLtd/firmware/tree/main/StarLite/MkV/coreboot/25.02, cabextracted it, then `sudo flashrom -p internal -w 25.02.rom -i bios --ifd -n -N`, and my bios says 25.02.
> 
> How is it not released?
> 
> Edit: nvm, its not yet released on LVFS...

Can you connect to your BT mouse?

**Comment by tbocek** [comment]
I cannot connect my M650 nor my Keychron B1. I'm using Arch with 6.13.5, but I have not tried older kernels yet. Scanning for BT devices, shows for me 0 devices.

**Comment by Sean-StarLabs** [comment]
> Notice that from upgrade battery level shows 0%

You've flashed the wrong firmware

**Comment by Sean-StarLabs** [comment]
> Do you know how to fix it?

I'd assume you'd need this:
https://github.com/StarLabsLtd/firmware/raw/refs/heads/main/StarLite/MkV-SB/coreboot/25.02/coreboot-I5-SB.cab

**Comment by jothgard** [comment]
> > Do you know how to fix it?
> 
> I'd assume you'd need this: https://github.com/StarLabsLtd/firmware/raw/refs/heads/main/StarLite/MkV-SB/coreboot/25.02/coreboot-I5-SB.cab

Did install it now, bios shows 25.02, but bluetooth mouse still cant connect

**Comment by tbocek** [comment]
Also installed this rom i5-SB, still no devices found.

```
# dmidecode 3.6
Getting SMBIOS data from sysfs.
SMBIOS 3.0.0 present.
Table at 0x76980000.

Handle 0x0000, DMI type 0, 26 bytes
BIOS Information
        Vendor: coreboot
        Version: 25.02
        Release Date: 03/06/2025
        ROM Size: 16 MB
        Characteristics:
                PCI is supported
                PC Card (PCMCIA) is supported
                BIOS is upgradeable
                Selectable boot is supported
                ACPI is supported
                Targeted content distribution is supported
        BIOS Revision: 0.0
        Firmware Revision: 25.2

Handle 0x0001, DMI type 1, 27 bytes
System Information
        Manufacturer: Star Labs
        Product Name: StarLite
        Version: 1.0
        Serial Number: I500000
        UUID: Not Settable
        Wake-up Type: Reserved
        SKU Number: I5-SB
        Family: I5-SB
```

**Comment by Sean-StarLabs** [comment]
> I'll post when is it, along with some of the versions on the linux side that has bugs

Please try to avoid posting with every experience; lots of people subscribe to threads for updates and get annoyed with too many notifications that don't help.

After all, you're all flashing binaries that:
* we haven't even tested
*  installing them in an unsupported way (we don't want people just reading random comments in an issue, following steps that other people have done, and then ending up with bricks or other issues)
* that can be affected by 3 different Linux bugs

General etiquette for git issues is hitting the thumbs up for "me too".

Details on the bugs, how to avoid etc. will follow once we've finished our final testing and actually released this update.

**Comment by jpritikin** [comment]
> No, it wasn't - https://fwupd.org/lvfs/devices/com.starlabs.I5.coreboot

Oh! I'll watch that page instead of the github repo.

Thanks.

**Comment by Sean-StarLabs** [comment]
25.02 pushed to stable

Still bisecting but:
* bluez 5.77 will not work (5.76 and 5.78 both fine)
* most kernels after 6.11.1 will not work, problem originates from one of three commits:

7b05933340f4490ef5b09e84d644d12484b05fdf
"Bluetooth: btusb: Fix not handling ZPL/short-transfer"

cfbfeee61582e638770a1a10deef866c9adb38f5
"Bluetooth: hci_sync: Ignore errors from HCI_OP_REMOTE_NAME_REQ_CANCEL"

d47da6bd4cfa982fe903f33423b9e2ec541e9496
"Bluetooth: hci_core: Fix sending MGMT_EV_CONNECT_FAILED"

It diverges from there, as quite a few distro's have non-upstreamed patches to try and fix, but we're liaising with the Manjaro guys to get a patch upstreamed that'll cover 6.11.2+

**Comment by maltklaus** [comment]
>     * bluez 5.77 will not work (5.76 and 5.78 both fine)
> 
>     * most kernels after 6.11.1 will not work, problem originates from one of three commits:

Thanks for your efforts!
Results of my testing:
- updated to 25.02 (this time using official methods...)
- Manjaro-testing with bluez 5.79 (not sure if this should work or not)
- current 6.12 kernel: not working as expected; Kernel 6.6 LTS, which is prior to the kernel patches you mentioned: Also not working

**Comment by Sean-StarLabs** [comment]
> Manjaro-testing with bluez 5.79 (not sure if this should work or not)

Don't know - Manjaro/Arch behaves differently to others, not sure why yet

> which is prior to the kernel patches

Is that an assumption or verified?

**Comment by Sean-StarLabs** [comment]
> Hopefully this helps

Ha, yes, thank you - it would all fall into place if it was broken 6.5ish, fixed in 6.8/9, then broken again in 6.11 (ignoring bluez 5.77).

Narrows it down to about 50,000 commits, if 6.6 doesn't have backports :)

**Comment by adamtclark** [comment]
FWIW, I'm now updated to 25.02, installed via fwupdmgr. BLE still does not work (though everything connects fine if I use a dongle). Running the Ubuntu 24.04.2 LTS that came factory installed from Starlite.

@Sean-StarLabs - can you please advise on whether 25.02 was expected to fix BLE on Ubuntu - and if so, whether you are advising us to update bluez to 5.76 or 5.78? I'm a bit hesitant to do so, as I can at least get things to work with a dongle at the moment, and would prefer not monkeying around with things unless necessary.

System details:
cat /sys/class/dmi/id/bios_version -> 25.02
bluetoothd -v -> 5.72

fwupdmgr get-history ->
Star Labs StarLite
│
├─StarLite (bios):
│ │   Device ID:          dbee8bd3b1ae0316ad143336155651eedb495a0e
│ │   Previous version:   24.07
│ │   Install Duration:   55 seconds
│ │   Update State:       Success
│ │   Last modified:      2024-09-17
│ │   GUID:               b2e0b708-4ced-5edb-83fe-eac07c774b3a
│ │   Device Flags:       • Internal device
│ │                       • Updatable
│ │                       • System requires external power source
│ │                       • Supported on remote server
│ │                       • Reported to remote server
│ │                       • Needs shutdown after installation
│ │                       • Cryptographic hash verification is available
│ │ 
│ └─StarLite MkV coreboot System Update:
│       New version:      24.08
│       Remote ID:        lvfs
│       Release ID:       98493
│       Summary:          coreboot firmware for StarLite MkV
│       License:          Proprietary
│       Size:             16.8 MB
│       Created:          2024-09-06
│       Urgency:          Medium
│         Tested:         2024-09-06
│         Distribution:   ubuntu 24.04
│         Old version:    24.07
│         Version[fwupd]: 2.0.0
│       Vendor:           Star Labs
│       Description:      
│       This release includes the following features:
│       
│       • Add a new driver for WiFi
│       • Add a new driver for Bluetooth
│       • Adjust how the ROTM is reported
│       • Optimise initial GPIO configuration
│       • Improve battery detection
│       • Improve battery profile initialisation
│       • Make the touchscreen frequency configurable
│       • Remove superfluous PD messages
│       • Rebased on coreboot 24.08
│       • Rebased on edk2/master
│       Checksum:         28fee919279f662c41b4bc94ba6fd48bead9172bf87f2ded997f9f742d5dfa03
│     
└─StarLite (bios):
  │   Device ID:          074f77b052bcaab5b61f3d4cd014d7d75dac454d
  │   Previous version:   25.01
  │   Install Duration:   1 minute
  │   Update State:       Success
  │   Last modified:      2025-03-16
  │   GUID:               b2e0b708-4ced-5edb-83fe-eac07c774b3a
  │   Device Flags:       • Internal device
  │                       • Updatable
  │                       • System requires external power source
  │                       • Supported on remote server
  │                       • Needs shutdown after installation
  │                       • Cryptographic hash verification is available
  │                       • Can tag for emulation
  │ 
  └─StarLite MkV coreboot System Update:
        New version:      25.02
        Remote ID:        lvfs
        Release ID:       108944
        Summary:          coreboot firmware for StarLite MkV
        License:          Proprietary
        Size:             16.8 MB
        Created:          2025-03-06
        Urgency:          Medium
          Tested:         2025-03-07
          Distribution:   ubuntu 24.10
          Old version:    24.12
          Version[fwupd]: 1.9.24
        Vendor:           Star Labs
        Release Flags:    • Trusted payload
                          • Trusted metadata
        Description:      
        General Updates:
        
        • Added support for low power mode for wireless card
        • Enable waking up from suspend by pressing any key on the keyboard
        • Add an option to disable the fan
        • Add an option to completely ignore the lid switch
        • General adjustments to save power
        • [StarLite Mk V] Low power drain when using the magnetic keyboard
        
        Bug Fixes/Enhancements:
        
        • Improved detection for third-party SSDs
        • Use interrupts for the TPM instead of polling
        • Fixed an issue where Linux would hang trying to initialise the TPM
        • [LabTop Mk IV] Fixed delay when pressing brightness control keys
        • [StarLite Mk V] Fixed regression that stopped Bluetooth communication
        Checksum:         7c66ff3702a45d39e21d16bec8fdf90c5246c4ccb53d9b1c74d7134493967a05

**Comment by BlackIkeEagle** [comment]
Hi, running Arch linux and having the same issue with the LE device not working

Currently I have confirmed with all latest packages (stable) installed the LE bluetooth device works with

- 6.1.131 LTS kernel
- 6.10.14 (last 6.10 tag in the linux git tree)

And it is no longer working from 6.11rc1 for me

Strangely enough the earlier mentioned commits can't be found between 6.10.14 and 6.11rc1 (or I'm doing something wrong)

**Comment by Sean-StarLabs** [comment]
@adamtclark For Ubuntu (and derivatives) you can just grab 6.11.1 from mainline (`ppa:cappelikan/ppa`)

**Comment by adamtclark** [comment]
> [@adamtclark](https://github.com/adamtclark) For Ubuntu (and derivatives) you can just grab 6.11.1 from mainline (`ppa:cappelikan/ppa`)

Thanks - sadly still no dice. I added ppa:cappelikan/ppa to my repository list, installed mainline, used the GUI to install Linux Kernel 6.11.1, and then rebooted. Running "uname -r" now returns "6.11.1-061101-generic", but BLE devices still do not appear (unless I plug in a dongle, in which case they work without issue).

**Comment by Sean-StarLabs** [comment]
> sadly still no dice

I see the same - 24.10 with 6.11.1 works, 24.04 with 6.11.1 doesn't, so much be something else in the equation

**Comment by BlackIkeEagle** [comment]
Since github continues to refuse me to upload a file in the ticket, a gist:

https://gist.github.com/BlackIkeEagle/630e76164d9eca5f1eb617888c7f1576

**Comment by ThomasLobker** [comment]
Tested with Ubuntu 25.10 and various kernels via de Mainline Kernels GUI. The default kernel is `6.11.0-19.19` and it doesn't work. I tried kernels `6.11.1` and the latest `6.13.8` and both didn't work either. Then I tried the older `6.1.131` kernel and finally I was able to connect with my Bluetooth mouse. I had to disable secure boot because the kernels are not signed by Canonical.

It's a temporary workaround, but I hope a more permanent solution will be available quite soon too.

**Comment by BlackIkeEagle** [comment]
I have added my additional information to a bugzilla report on linux-bluetooth: https://bugzilla.kernel.org/show_bug.cgi?id=219553

Hopefully this results in a proper fix in upstream linux

**Comment by BlackIkeEagle** [comment]
@Sean-StarLabs Don't know if the bisect and "workaround" patch is of any help to find an actual solution for this bluetooth issue?

**Comment by Sean-StarLabs** [comment]
It was, very - just waiting to hear back from an Intel dev about it, as I think, rather than reverting, we can just guard the change

**Comment by rogercreagh** [comment]
Since I have a bluetooth mouse and keyboard and headphones working with firmware version 24.10 and ubuntu up-to-date I am assuming that it is not safe to update the firmware until this issue is finally marked resolved. Is this correct?

**Comment by Sean-StarLabs** [comment]
>  Is this correct?

No, issue is kernel (save for firmware 25.01)

**Comment by seindal** [comment]
I ran into this with a Logitech mouse, pebble m350s. I had it working earlier.

With kernel 6.12 no go, even with downgraded firmware as per the first post on this issue.

With kernel 6.0.32 from Debian stable, it pairs and works.

```
Mar 31 10:30:39 star kernel: bluetooth hci0: firmware: direct-loading firmware intel/ibt-0040-2120.sfi
Mar 31 10:30:39 star kernel: Bluetooth: hci0: Found device firmware: intel/ibt-0040-2120.sfi
Mar 31 10:30:39 star kernel: Bluetooth: hci0: Firmware Version: 151-5.24
```

Starlite mkV, Debian testing, firmware from package firmware-iwlwifi version 20241210-1

**Comment by dchang0** [comment]
I can confirm BlackIkeEagle's findings indepedently. I submitted a ticket to Ubuntu (but the bug is in the kernel) here:

https://bugs.launchpad.net/ubuntu/+bug/2100565

While BlackIkeEagle's patch will work, I am not sure we should simply roll back Ying Hsu's overbroad re-classification of packets. I suspect that the real fix is to discriminate between types of packets on the Jefferson Peak bluetooth controller--some packets might need to be reclassified (probably those to LE audio devices), others might not (such as mice and keyboard traffic).

I'm going to wait for the kernel people to figure out a more precise solution with Intel's expertise.

**Comment by oscaropenness** [comment]
Is there any news on this issue?
I am still stuck on Kernel 6.10.10 so I can use my mouse with the Starlite.
Can I easily include the fix from BlackIkeEagle in a custom Kernel on my ArchLinux?

**Comment by BlackIkeEagle** [comment]
> Is there any news on this issue? I am still stuck on Kernel 6.10.10 so I can use my mouse with the Starlite. Can I easily include the fix from BlackIkeEagle in a custom Kernel on my ArchLinux?

@oscaropenness https://gitlab.com/herecura/packages/linux-bede/-/blob/0c26f6493c05ec968ce572cf33661b7a2a0d1151/0002-Bluetooth-btintel-don-t-reclassify-signal-for-GfP2-a.patch should work just fine, applies on 6.14 and 6.15 (I did not yet personally upgrade to 6.15 on my starlite but my guess is it still works as expected)

**Comment by BlackIkeEagle** [comment]
> > > Is there any news on this issue? I am still stuck on Kernel 6.10.10 so I can use my mouse with the Starlite. Can I easily include the fix from BlackIkeEagle in a custom Kernel on my ArchLinux?
> > 
> > 
> > [@oscaropenness](https://github.com/oscaropenness) https://gitlab.com/herecura/packages/linux-bede/-/blob/0c26f6493c05ec968ce572cf33661b7a2a0d1151/0002-Bluetooth-btintel-don-t-reclassify-signal-for-GfP2-a.patch should work just fine, applies on 6.14 and 6.15 (I did not yet personally upgrade to 6.15 on my starlite but my guess is it still works as expected)
> 
> running kernel 6.14.8, seems still not find my Bluetooth mouse.

custom build based on https://github.com/archlinux/linux with the extra patch? In my own build the extra bluetooth patch from the arch kernel is missing since I don't seem to be affected.

**Comment by DevenBL** [comment]
I applied the patch and it allows ble devices to connect and be detected. However they disconnect and then reconnect periodically.

Kernel: 6.15.4-gentoo-dist-hardened
Linux-firmware 20250410

dmesg log
[ 5052.871392] Bluetooth: hci0: Hardware error 0x0c
[ 5052.887502] Bluetooth: hci0: Exception info 
[ 5058.862616] input: MX Anywhere 3S Mouse as /devices/virtual/misc/uhid/redacted/input/input59
[ 5058.863916] hid-generic redacted: input,hidraw3: BLUETOOTH HID v0.03 Mouse [MX Anywhere 3S] on redacted

[   15.666562] Bluetooth: Core ver 2.22
[   15.666606] NET: Registered PF_BLUETOOTH protocol family
[   15.666607] Bluetooth: HCI device and connection manager initialized
[   15.666612] Bluetooth: HCI socket layer initialized
[   15.666614] Bluetooth: L2CAP socket layer initialized
[   15.666618] Bluetooth: SCO socket layer initialized
[   15.797346] usbcore: registered new interface driver btusb
[   15.799442] Bluetooth: hci0: Device revision is 2
[   15.799449] Bluetooth: hci0: Secure boot is enabled
[   15.799451] Bluetooth: hci0: OTP lock is enabled
[   15.799452] Bluetooth: hci0: API lock is enabled
[   15.799453] Bluetooth: hci0: Debug lock is disabled
[   15.799455] Bluetooth: hci0: Minimum firmware build 1 week 10 2014
[   15.799457] Bluetooth: hci0: Bootloader timestamp 2019.40 buildtype 1 build 38
[   15.799582] Bluetooth: hci0: DSM reset method type: 0x00
[   15.799653] Loading firmware: intel/ibt-0040-2120.sfi
[   15.805701] Bluetooth: hci0: Found device firmware: intel/ibt-0040-2120.sfi
[   15.805748] Bluetooth: hci0: Boot Address: 0x100800
[   15.805750] Bluetooth: hci0: Firmware Version: 151-5.24

[   17.346941] Bluetooth: hci0: Waiting for firmware download to complete
[   17.347379] Bluetooth: hci0: Firmware loaded in 1505534 usecs
[   17.347544] Bluetooth: hci0: Waiting for device to boot
[   17.364462] Bluetooth: hci0: Device booted in 16546 usecs
[   17.366511] Loading firmware: intel/ibt-0040-2120.ddc
[   17.366709] Bluetooth: hci0: Found Intel DDC parameters: intel/ibt-0040-2120.ddc
[   17.368490] Bluetooth: hci0: Applying Intel DDC parameters completed
[   17.371547] Bluetooth: hci0: Firmware timestamp 2024.5 buildtype 1 build 151
[   17.371556] Bluetooth: hci0: Firmware SHA1: 0x00000000
[   17.375493] Bluetooth: hci0: Fseq status: Success (0x00)
[   17.375500] Bluetooth: hci0: Fseq executed: 00.00.02.36
[   17.375504] Bluetooth: hci0: Fseq BT Top: 00.00.02.36

[   23.094924] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   23.094929] Bluetooth: BNEP filters: protocol multicast
[   23.094934] Bluetooth: BNEP socket layer initialized
[   23.096292] Bluetooth: MGMT ver 1.23

[   35.113171] Bluetooth: RFCOMM TTY layer initialized
[   35.113182] Bluetooth: RFCOMM socket layer initialized
[   35.113191] Bluetooth: RFCOMM ver 1.11

**Comment by Sean-StarLabs** [comment]
Fixed merged upstream - https://github.com/torvalds/linux/commit/6ec3185fbc3528f2284c347fb9bd8be6fa672ed4

**Comment by seindal** [comment]
I don't think the fix made it into 6.16. I tried with the Debian packaged kernel, and it doesn't find anything of the LE devices I have.

**Comment by gareth-anthony-hulse** [comment]
Cannot pair the StarLabs StarLite MkV with the Keychron K1 QMK Wireless Mechanical Keyboard (ISO Layout) - Version 6, which is the UK keyboard layout. The keyboard doesn't have any issue connecting to my Arch Linux desktop.

**Specs**:https://linux-hardware.org/?probe=5c8e537528

My old Sony WH-1000XM3 pairs to the StarLite just fine though.
